packages/block-editor/src/store/selectors.js

```diff
@@ -2037,16 +2038,70 @@ export const getInserterItems = createRegistrySelector( ( select ) =>
 				buildScope: 'inserter',
 			} );
 
-			const blockTypeInserterItems = getBlockTypes()
+			let blockTypeInserterItems = getBlockTypes()
 				.filter( ( blockType ) =>
-					canIncludeBlockTypeInInserter(
-						state,
-						blockType,
-						rootClientId
-					)
+					hasBlockSupport( blockType, 'inserter', true )
 				)
 				.map( buildBlockTypeInserterItem );
 
+			if ( options[ withRootClientIdOptionKey ] ) {
```

Comment:
I think adding a comment of how the discovery algorithm works would be great to read in a couple of months :D